### PR TITLE
feat: return unresolved dependencies from detector

### PR DESCRIPTION
**Pull request by obycode**

### CODE REVIEW

File: src/analysis/ast_dependency_detector.rs [hunk 1]
```diff
@@ -46,14 +48,105 @@ pub struct ASTDependencyDetector<'a> {
...
 
+#[derive(Clone, Debug, Eq)]
+pub struct Dependency {
+    pub contract_id: QualifiedContractIdentifier,
+    pub required_before_publish: bool,
+}
+
+impl PartialEq for Dependency {
+    fn eq(&self, other: &Self) -> bool {
+        self.contract_id == other.contract_id
```
Comment (lgalabru) [inline src/analysis/ast_dependency_detector.rs:63]: Shouldn't we including `required_before_publish` in this equality?

Comment (obycode) [reply src/analysis/ast_dependency_detector.rs:63]: See the `add_dependency` method. What I was going for is that a contract is only in the set once (regardless of this flag), and a dependency with the `required_before_publish` flag overwrites a dependency without that flag. Is this a confusing way to do that?

Comment (lgalabru) [reply src/analysis/ast_dependency_detector.rs:63]: Works for me, thank you unpacking!

**Review by lgalabru — APPROVED** [review]
LGTM!

**Pull request merged**